src/hotspot/share/runtime/continuationFreezeThaw.cpp

```diff
@@ -1988,34 +1991,34 @@ void ThawBase::recurse_thaw_compiled_frame(const frame& hf, frame& caller, int n
     _cont.tail()->do_barriers<stackChunkOopDesc::BarrierType::Store>(_stream, SmallRegisterMap::instance);
   }
 
-  const bool bottom = recurse_thaw_java_frame<ContinuationHelper::CompiledFrame>(caller, num_frames);
+  const bool is_bottom_frame = recurse_thaw_java_frame<ContinuationHelper::CompiledFrame>(caller, num_frames);
 
-  DEBUG_ONLY(before_thaw_java_frame(hf, caller, bottom, num_frames);)
+  DEBUG_ONLY(before_thaw_java_frame(hf, caller, is_bottom_frame, num_frames);)
 
   assert(caller.sp() == caller.unextended_sp(), "");
 
-  if ((!bottom && caller.is_interpreted_frame()) || (bottom && Interpreter::contains(_cont.tail()->pc()))) {
+  if ((!is_bottom_frame && caller.is_interpreted_frame()) || (is_bottom_frame && Interpreter::contains(_cont.tail()->pc()))) {
     _align_size += frame::align_wiggle; // we add one whether or not we've aligned because we add it in freeze_interpreted_frame
   }
 
-  frame f = new_stack_frame<ContinuationHelper::CompiledFrame>(hf, caller, bottom);
-  intptr_t* const frame_sp = f.sp();
-  intptr_t* const heap_sp = hf.unextended_sp();
+  frame f = new_stack_frame<ContinuationHelper::CompiledFrame>(hf, caller, is_bottom_frame);
+  intptr_t* const frame_top = f.sp();
+  intptr_t* const heap_top = hf.unextended_sp();
 
-  const int added_argsize = (bottom || caller.is_interpreted_frame()) ? hf.compiled_frame_stack_argsize() : 0;
+  const int added_argsize = (is_bottom_frame || caller.is_interpreted_frame()) ? hf.compiled_frame_stack_argsize() : 0;
   int fsize = ContinuationHelper::CompiledFrame::size(hf) + added_argsize;
   assert(fsize <= (int)(caller.unextended_sp() - f.unextended_sp()), "");
 
-  intptr_t* from = heap_sp - frame::metadata_words;
-  intptr_t* to   = frame_sp - frame::metadata_words;
+  intptr_t* from = heap_top - frame::metadata_words;
+  intptr_t* to   = frame_top - frame::metadata_words;
```

Comment:
Some names could be more explicit.
For instance, instead of frame_top and heap_top, using stack_frame_top and heap_frame_top would be more precise, making explicit that the top value refers to a frame, which is either still on the stack, or being stored in the heap.